### PR TITLE
libs/libc/net: implement socketpair

### PR DESCRIPTION
**Pull request by Dan-guanghua**
Reference here:
https://www.freebsd.org/cgi/man.cgi?apropos=0&sektion=2&query=socketpair&manpath=FreeBSD+7.0-current&format=html

Change-Id: I959b4d661cd436f5d5050baf9e821db84139d482
Signed-off-by: danguanghua <danguanghua@xiaomi.com>

### CODE REVIEW

File: libs/libc/net/lib_socketpair.c [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    }
+}
+
+static int create_socket(int domain, int type, int protocol,
+                         FAR union sockaddr_u *addr, FAR socklen_t *len)
+{
+  int socketfd;
+
+  socketfd = socket(domain, type, protocol);
+  if (socketfd < -1)
```
Comment (Ouss4) [inline libs/libc/net/lib_socketpair.c:75]:
```suggestion
  if (socketfd < 0)
```

Comment (Dan-guanghua) [reply libs/libc/net/lib_socketpair.c]: > 

done

File: libs/libc/net/lib_socketpair.c [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+          goto err;
+        }
+
+      sv[0] = accept(listener, &addr[0].addr, &len);
+      if (sv[0] < 0)
+        {
+          goto err;
+        }
+
+      close(listener);
```
Comment (xiaoxiang781216) [inline libs/libc/net/lib_socketpair.c:170]: move before line 165

Comment (Dan-guanghua) [reply libs/libc/net/lib_socketpair.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**